crates/tx5-connection/src/conn.rs

```diff
@@ -238,12 +238,28 @@ impl TaskCore {
             .fetch_add(len as u64, Ordering::Relaxed);
     }
 }
+/// Drop guard that closes the ready semaphore. This ensures callers will not be waiting
+/// for ready indefinitely after the task has ended.
+struct DropReady {
+    ready: Arc<tokio::sync::Semaphore>,
+}
+
+impl Drop for DropReady {
+    fn drop(&mut self) {
+        self.ready.close();
```

Comment:
I don't think it's acceptable to close this if a ready state is not reached. Closing the ready semaphore allows the code to proceed as though the connection can be used and we then rely on something downstream of the ready check failing and allowing everything to shut down. If nothing else, that is hard to reason about when reading the code.

The failure to ready is already handled in https://github.com/holochain/tx5/blob/main/crates/tx5/src/backend/be_tx5_connection.rs#L58 which will catch the case that something fails. It's not ideal and not as sophisticated as the solution higher up the stack. Changing that though, I think should look like communicating success or failure with the semaphore close, and not just closing it and letting the code proceed as though the connection is ready.